lib/fluent/plugin/in_http.rb

```diff
@@ -202,11 +202,17 @@ def on_request(path_info, params)
             if @add_remote_addr
               single_record['REMOTE_ADDR'] = params['REMOTE_ADDR']
             end
-            single_time = if t = single_record.delete('time')
-                            Fluent::EventTime.from_time(Time.at(t))
-                          else
-                            time
-                          end
+
+            if single_record.has_key?('time')
+                single_time = Fluent::EventTime.from_time(Time.at(single_record['time']))
+            else
+                single_time = time
+            end
+
+            unless @parser and @parser.keep_time_key
```

Comment:
This condition can be moved to under above `if single_record._has_key?`

Reply:
I fixed this point by 39f3a0da